scripts/o3de/o3de/repo_properties.py

```diff
@@ -0,0 +1,399 @@
+        FileNotFoundError: If the json_data_path does not exist.
+        ValueError: If the json_data_path is not a dict.
+    """
+    # basename = f'{object_name}_release_1.0.0'
+    # zip_filename = f'{archive_filename}.zip'
+    basename = archive_filename.stem
+    zip_filename = archive_filename
+
+    json_data = merge_json_data(json_data_path, {
+        'repo_uri':repo_uri,
+        'download_source_uri':f'{download_prefix}/{zip_filename}',
+    })
+
+    logging.info(f"Creating '{releases_path / zip_filename}'")
+
+    # create the release zip file
+    shutil.make_archive(releases_path / basename, 'zip', src_data_path)
+
+    zip_path = releases_path / zip_filename
+
+    with zip_path.open('rb') as f:
+        json_data['sha256'] = hashlib.sha256(f.read()).hexdigest()
+
+    return json_data
+
+# retrieves json data of the repository from a file path
+def get_repo_props(path: pathlib.Path) -> dict:
+    repo_json = manifest.get_json_data_file(path, "Repo", validation.valid_o3de_repo_json)
+    if not isinstance(repo_json, dict):
+        logger.error(f'Could not retrieve repo.json file for {path}')
+        return None
+    return repo_json
+
+
+def _find_index(data:list, key:str, value:str) -> int:
+    for index, item in enumerate(data):
+        if item.get(key) == value:
+            return index
+    return -1
+
+def _changed(original:dict, new:dict) -> dict:
+    changed = {}
+    for key, value in new.items():
+        if key not in original:
+            changed[key] = value
+        elif original[key] != value:
+            changed[key] = value
+
+    return changed
+
+def _edit_objects(object_typename:str,
+                  validator:callable,
+                  repo_json: dict,
+                  add_objects: pathlib.Path or list = None,
+                  delete_objects: str or list = None,
+                  replace_objects: pathlib.Path or list = None,
+                  release_archive_path: pathlib.Path = None,
+                  download_prefix: str = None):
+    """
+    Edits and modifies the 'gem_data/projects_data/template_data' in the repo_json parameter
+    :param object_typename: The type object field you want to change
+    :param validator: validates the object you want to edit
+    :param add_objects: Any object paths you want to add to your repo_json object field
+    :param delete_objects: Any object_names to be removed from your repo_json object field
+    :param replace_objects: A list of object paths that will completely replace the current object_list
+    :param release_archive_path: Path where you want your release to be located
+    :param download_prefix: The prefix of the download uri
+    """
+    # The beginning remote repo json template
+    repo_objects_data = repo_json.get(f'{object_typename}s_data',[])
+
+    if add_objects:
+        paths = add_objects.split() if isinstance(add_objects, str) else add_objects
+        for object_path in paths:
```

Comment:
If looks like replace_objects is just like add_objects but cleears the repo_objects_data first so you could just do this:
```suggestion
    if add_objects or replace_objects:
        if replace_objects:
            repo_objects_data = []
            paths = replace_objects.split() if isinstance(replace_objects, str) else replace_objects
        else:
            paths = add_objects.split() if isinstance(add_objects, str) else add_objects
            
        for object_path in paths:
```